src/commands/importExportEntities/entities/languageVariants.ts

```diff
@@ -15,18 +23,300 @@ export const languageVariantsExportEntity: EntityDefinition<
         .listLanguageVariantsByCollection()
         .byCollectionCodename(collection.codename)
         .toAllPromise()
-        .then(res => res.data.items)
+        .then(res => res.responses.flatMap(res => res.rawData.variants))
     );
 
     const variants = await serially(promises);
 
     return variants.flatMap(arr => arr);
   },
-  serializeEntities: collections => JSON.stringify(collections),
-  importEntities: () => {
-    throw new Error("Not supported yet.");
-  },
-  deserializeEntities: () => {
-    throw new Error("Not supported yet.");
+  serializeEntities: JSON.stringify,
+  deserializeEntities: JSON.parse,
+  importEntities: async (client, fileVariants, context) => {
+    await serially(fileVariants.map(createImportVariant(client, context)));
   },
 };
+
+const createImportVariant =
+  (client: ManagementClient, context: ImportContext) =>
+  (fileVariant: LanguageVariantContracts.ILanguageVariantModelContract) =>
+  async (): Promise<true> => {
+    const typeContext = context.contentTypeContextWithElementsByOldIds
+      .get(context.contentItemContextByOldIds.get(fileVariant.item.id ?? "")?.oldTypeId ?? "");
+
+    if (!typeContext) {
+      throw new Error(`Cannot find type context for item "${fileVariant.item.id}".`);
+    }
+
+    const newWfContext = findTargetWfStep(context, fileVariant);
+
+    const projectVariant = await client
+      .upsertLanguageVariant()
+      .byItemId(context.contentItemContextByOldIds.get(fileVariant.item.id ?? "")?.selfId ?? "")
+      .byLanguageId(context.languageIdsByOldIds.get(fileVariant.language.id ?? "") ?? "")
```

Comment:
Well, the best thing to do would be to improve the types. The next best thing is to throw an exception and then probably the `as` cast. 😄 We use this approach in many places across the data-ops codebase (e.g. a variant also always has an item as well... I'd like to solve this systematically instead of fixing this one place. I am thinking about creating a proper type and casting to it (e.g. `as BetterReference`) and using it across the codebase. Then, once we fix the SDK, we can just delete the type and have it fixed across the whole solution once again. What do you think?